### ISSUE TITLE
Change from ps2epsi to eps2eps (in pull request #7496) causes spacing problems

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#7578]
I'm using gs version 9.27
dvips version 5.999 (kpathsea version 6.3.1), comes with texlive 2019 (macosx)

Input with standard config file:

    /*! \mainpage
     *
     *  This is \f$x\f$ and \f$y\f$.
     */

Gives this output on master:
<img width="311" alt="Screenshot 2020-02-09 at 17 23 16" src="https://user-images.githubusercontent.com/1300762/74106987-f60f4080-4b6b-11ea-8fa7-744e516a2c99.png">

and this output on 1.8.17:
<img width="321" alt="Screenshot 2020-02-09 at 17 23 53" src="https://user-images.githubusercontent.com/1300762/74106990-fc052180-4b6b-11ea-9172-470d0c5f80c5.png">

Note the extra spacing in the master version.

Example with output for both 1.8.17 and master including all temporary files that are normally deleted:
[eps2eps_issue.zip](https://github.com/doxygen/doxygen/files/4177263/eps2eps_issue.zip)

**Comment by albert-github** [comment]
I did a number of tests with the ghostscript versions available (under the hood `eps2eps` and `ps2epsi` use again `gs` / `gswin32c` / `mgs`. `mgs` is the ghostscript version as used by MikTeX, at least as far as I could see).

Results:

**eps2eps** i.e. command like: 
`"%GS_EXECUTABLE%" -q -sDEVICE=epswrite "-sOutputFile=form0_%VERSION%.eps" -dNOPAUSE -dBATCH -P- -dNOSAFER _form0.eps`
```
form0_9.05.eps:%%BoundingBox: 149 657 154 662
form0_9.05.eps:%%Creator: GPL Ghostscript 905 (epswrite)

form0_9.50.eps:%%BoundingBox: 148 654 160 665
form0_9.50.eps:%%Creator: GPL Ghostscript 950 (eps2write)

form0_cygwin.eps:%%BoundingBox: 148 654 160 665
form0_cygwin.eps:%%Creator: GPL Ghostscript 927 (eps2write)

form0_miktex.eps:%%BoundingBox: 149 657 155 662
form0_miktex.eps:%%Creator: MiKTeX GPL Ghostscript 925 (eps2write)
```
So looks a bit like that after version 9.27 the BoundingBox is enlarged (in our case nearly doubled).

**ps2epsi**, command like:
`"%GS_EXECUTABLE%" -q -dBATCH -dNOPAUSE -P- -DNOSAFER -sDEVICE=bbox _form0.eps 2> form0_cygwin_ps2epsi.eps`
```
form0_9.05_ps2epsi.eps:%%BoundingBox: 148 657 154 662

form0_9.50_ps2epsi.eps:%%BoundingBox: 148 657 154 662

form0_cygwin_ps2epsi.eps:%%BoundingBox: 144 653 149 658

form0_miktex_mgs_ps2epsi.eps:%%BoundingBox: 148 657 154 662

form0_miktex_ps2epsi.eps:%%BoundingBox: 148 657 154 662
form0_miktex_ps2epsi.eps:%%BoundingBox: 149 657 155 662
form0_miktex_ps2epsi.eps:%%Creator: MiKTeX GPL Ghostscript 925 (eps2write)
```
Looks like all the versions give good results. 

@doxygen:
- Maybe we should remove the call to `eps2eps` and replace it directly with the "call": `"%GS_EXECUTABLE%" -q -dBATCH -dNOPAUSE -P- -DNOSAFER -sDEVICE=bbox _form0.eps 2> form0_cygwin_ps2epsi.eps`

  I don't think it is advisable to call `ps2epsi` due to the problem with with the `-dNOSAFER` option that is not supported there.

**Comment by doxygen** [comment]
@albert-github  With [this commit](https://github.com/doxygen/doxygen/commit/10b2b8fc694b60a17ccd2642f3a40c851e33b9da) the formula handling has been reimplemented. I took your advise and replaced `eps2eps` and `ps2epsi` by using `ghostscript` only.

I also improve rendering of the bitmaps for the HTML output (higher resolution, better anti-aliasing, explicit size in the output for faster rendering) and added an option `HTML_FORMULA_FORMAT` to allow creation of SVG images (using [pdf2svg](https://github.com/dawbarton/pdf2svg)).

**Comment by albert-github** [comment]
I had a look at the output with Windows and Cygwin.

- Windows with ghostscript version 9.05 has a problem as here the device `eps2write` does not yet exist but `epswrite` exists, I'm wondering why using `eps2write` and not `bbox`.
- Windows  with ghostscript version 9.50 results look OK to me (in attachment results in hw_50)
- Windows with the mgs from MikTex (i.e. version 9.25 and renamed to gswin32c) looks OK to me.
- Cygwin with ghostscript version 9.27 the images look more or less empty (only 2 unique colors present) (also running this executable gives the empty results. On Cygwin I can only find this version and the version 9.50 but the later is marked as Test.

Example plus results: [example.tar.gz](https://github.com/doxygen/doxygen/files/4198302/example.tar.gz)

**Comment by albert-github** [comment]
Regarding the SVG possibility, is indeed be a nice extension.

The pdf2svg repository looks like a bit of abandoned, wouldn't Inkscape be a better choice? The Inkscape package is also suggested for use with the proposed patch #7241

I cloned the https://github.com/jalios/pdf2svg-windows repository to obtain the Windows executables and set the path to the `pdf2svg` executable, but neither the 32-bit not the 64-bit version could be started.

Regarding Cygwin I don't see a default package (the required poppler and Cairo packages look like to be available). A quick test revealed that the `pdf2svg` can be easily build and gives the right results (I can also use this executable on Windows).

Currently the SVG format is only supported for HTML, why not supporting it for docbook as well (here e.g. call graphs in svg format do work as well). RTF dosn't (at the moment) make much sense as call graphs in svg format don't work either.  For LaTeX the native handling is superior and for XML it is left to the post processing step how to display the formula so no action required.

**Comment by doxygen** [comment]
> - Windows with ghostscript version 9.05 has a problem as here the device `eps2write` does not yet exist but `epswrite` exists, I'm wondering why using `eps2write` and not `bbox`.

It is because my version of ghostscript (9.27) does not have `epswrite` only `eps2write` and `bbox` only outputs the bounding box, nothing else. So older versions of ghostscript do not seem to work anymore at the moment ☹️ 
@albert-github  could you try if epswrite for gs 9.05 gives correct results? then maybe we can can do a version dependent call (ugh))

**Comment by albert-github** [comment]
I don't think it is worth to to check it  (I can do it though) as the 9.05 version is from 2012 and the current version is 9.50. and I don't like the version dependent workaround either. The bounding box (`bbox`) needs the work around of reading and adding.

Strange that you didn't have the problems I had with the 9.27 version ("empty images"), might be that the bounding box for the 9.27 is a bit strange(I saw also in the doxygen LaTeX manual some half formulas), but as said as 9.50 (though there was a big jump from 9.27 to 9.50) does work for me (Windows and Cygwin).

**Comment by doxygen** [comment]
> The pdf2svg repository looks like a bit of abandoned, wouldn't Inkscape be a better choice?

I tried (see also the development switches in `formula.cpp` in the latest commit if you want to try it out for yourself). For me it didn't always produce proper SVG files.

> Strange that you didn't have the problems I had with the 9.27 version ("empty images")

I saw these as well at some points during development, but it always turned out to be "stale" files that weren't regenerated. Cleaning the output directory and rerunning doxygen then fixed it again.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.18.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by TheGrumpyCoda** [comment]
Thank you for implementing the SVG support!
In case someone else also experiences issues caused by an upgrade to this newer version:

I upgraded from Doxygen 1.8.15 to 1.8.18. I'm using Windows with MiKTeX 2.9.7029, GhostScript 9.27. 
When generating the documentation for my project, all rendered formulas where offset and cut off like the following one:
![form_1_9 27](https://user-images.githubusercontent.com/60060564/83486563-b7e6b580-a4a9-11ea-84f9-a5165fefd49c.png)

To fix this, GhostScript had to be upgraded to GhostScript 9.52. After the update the formulas are rendered as expected using Doxygen 1.8.18:
![form_1_9 52](https://user-images.githubusercontent.com/60060564/83486582-c3d27780-a4a9-11ea-8f5d-aeb1bcc93a90.png)
